src/fidesops/api/deps.py

```diff
@@ -1,11 +1,17 @@
 from typing import Generator
 
+from fidesops.common_exceptions import FunctionalityNotConfigured
+from fidesops.core.config import config
 from fidesops.db.session import get_db_session
 from fidesops.util.cache import get_cache as get_redis_connection
 
 
 def get_db() -> Generator:
     """Return our database session"""
+    if not config.database.ENABLED:
+        raise FunctionalityNotConfigured(
```

Comment:
Let's add a tests for this and the `get_cache`. I'm thinking add a new file at `tests/api/test_deps.py` unless you think there is a better place to put it.

```py
import pytest

from fidesops.api.deps import get_cache, get_db
from fidesops.common_exceptions import FunctionalityNotConfigured
from fidesops.core import config


@pytest.fixture
def mock_config():
    db_enabled = config.config.database.ENABLED
    redis_enabled = config.config.redis.ENABLED
    config.config.database.ENABLED = False
    config.config.redis.ENABLED = False
    yield
    config.config.database.ENABLED = db_enabled
    config.config.redis.ENABLED = redis_enabled


@pytest.mark.usefixtures("mock_config")
def test_get_cache_not_enabled():
    with pytest.raises(FunctionalityNotConfigured):
        next(get_cache())


@pytest.mark.usefixtures("mock_config")
def test_get_db_not_enabled():
    with pytest.raises(FunctionalityNotConfigured):
        next(get_db())
```

Reply:
@sanders41 thank you for writing the test up! i ran a bit further with this and also added in a test on the endpoint layer to make sure the exception handlers work as expected - let me know what you think.